### PR TITLE
Upgrade to Spring Boot 3.2 (#2503)

### PR DESCRIPTION
**Pull request by glebfox**
See #2503

### CODE REVIEW

File: jmix-data/eclipselink/src/main/java/io/jmix/eclipselink/impl/JmixEclipselinkJtaTransactionManager.java [hunk 1]
```diff
@@ -20,8 +20,9 @@
 import jakarta.transaction.TransactionManager;
 import jakarta.transaction.UserTransaction;
 import org.eclipse.persistence.internal.helper.JmixUtil;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
 import org.springframework.transaction.TransactionDefinition;
-import org.springframework.transaction.support.DefaultTransactionStatus;
 
 public class JmixEclipselinkJtaTransactionManager extends JmixJtaTransactionManager {
```
Comment (glebfox) [inline jmix-data/eclipselink/src/main/java/io/jmix/eclipselink/impl/JmixEclipselinkJtaTransactionManager.java:27]: Unused

File: jmix-data/eclipselink/src/main/java/io/jmix/eclipselink/impl/JmixEclipselinkJtaTransactionManager.java [hunk 2]
```diff
@@ -31,19 +32,18 @@ public JmixEclipselinkJtaTransactionManager(String key,
         super(key, userTransaction, transactionManager);
     }
 
+    @Override
+    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
```
Comment (glebfox) [inline jmix-data/eclipselink/src/main/java/io/jmix/eclipselink/impl/JmixEclipselinkJtaTransactionManager.java:36]: Can't guarantee that this is 100% replacement of `prepareSynchronization` overriding

Comment (gorbunkov) [reply jmix-data/eclipselink/src/main/java/io/jmix/eclipselink/impl/JmixEclipselinkJtaTransactionManager.java:36]: Looks good

File: jmix-data/eclipselink/src/main/java/io/jmix/eclipselink/impl/JmixEclipselinkJtaTransactionManager.java (hunk 1 above)
Comment (gorbunkov) [reply jmix-data/eclipselink/src/main/java/io/jmix/eclipselink/impl/JmixEclipselinkJtaTransactionManager.java:27]: This class is used by a global transactions mechanism. We have an article about this in YT KB.

File: jmix-data/eclipselink/src/main/java/io/jmix/eclipselink/impl/JmixEclipselinkTransactionExecutionListener.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    private final String key;
+    private final ApplicationContext applicationContext;
+
+    public JmixEclipselinkTransactionExecutionListener(String key, ApplicationContext applicationContext) {
+        this.key = key;
+        this.applicationContext = applicationContext;
+    }
+
+    @Override
+    public void afterBegin(TransactionExecution transaction, Throwable beginFailure) {
```
Comment (gorbunkov) [inline jmix-data/eclipselink/src/main/java/io/jmix/eclipselink/impl/JmixEclipselinkTransactionExecutionListener.java:35]: IntelliJ Idea asks for a `@Nullable` annotation near the `Throwable beginFailure` parameter. Please, add.

File: jmix-gradle-plugin/build.gradle [hunk 4]
```diff
@@ -29,7 +29,7 @@ repositories {
 dependencies {
     implementation localGroovy()
 
-    implementation 'org.springframework.boot:spring-boot-gradle-plugin:3.1.5'
+    implementation 'org.springframework.boot:spring-boot-gradle-plugin:3.2.0'
```
Comment (gorbunkov) [inline jmix-gradle-plugin/build.gradle:32]: They have released the 3.2.1 version. Please upgrade to it.

**Review by gorbunkov — APPROVED** [review]

**Pull request merged**